service/collector.go

```diff
@@ -244,10 +241,9 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 // Run starts the collector according to the given configuration given, and waits for it to complete.
 // Consecutive calls to Run are not allowed, Run shouldn't be called once a collector is shut down.
 func (col *Collector) Run(ctx context.Context) error {
-	col.zPagesSpanProcessor = zpages.NewSpanProcessor()
 	col.tracerProvider = sdktrace.NewTracerProvider(
 		sdktrace.WithSampler(internal.AlwaysRecord()),
```

Comment:
This is needed only for the `zPagesSpanProcessor`. If not registered, we don't need to locally record all the spans.

Reply:
Somehow... there is something I don't like here ... (sorry, most likely is not this code, but the overall structure has some problems).

An idea that I had in mind for long time, is to actually embed the "zpages" into Host:
* Host will have to start an "http.Server" that can be shared between "health check", "zpages", "status", "etc.".
* If we do this, the zpages/health extensions (maybe) becomes almost a no-op.

How I would design this is probably on the Host interface allow any component to "Register" a path/handler similar with `http.ServerMux`, something like:
```golang
RegisterHandler(pattern string, handler http.Handler)
```

Then we should extend the "service" configuration to accept an http server configuration for this, and maybe some options to say "register trace zpages", "register metrics zpages", etc.

The `service.Host` can accept this instance of the `http.Server` (similar to how you construct things like telemetry).

Reply:
Let's focus on the zpages case first, since I am having a bit of trouble visualizing how this would work. There are two parts for zpages:

1. `RegisterZPages`, which needs to live in the service/host since it needs to know about the available pipelines
2. the `/debug/tracez` handler, which can be dealt within the `zpages` extension code or within service/host code

In this case, what you would want is to: for (1), change it so that you don't need to pass the `http.ServeMux` but rather have this be done within the service code (you would still need a special method on the component.Host to activate this, right?) and for (2) call the new `RegisterHandler` method from `component.Host` to register the `tracez` handler.

I guess the part that confuses me the most is 
> maybe some options to say "register trace zpages", "register metrics zpages", etc.

since I don't understand where these options live (do they live in `service.svcSettings`? are these methods part of `service.Host` but not part of the `component.Host` interface?)

Reply:
TL;DR: Who owns the "http.Server" instance. right now that is the main reason why we have the `zpages` extension, if we change the `serviceHost` to have a  default (can be shared between components) `http.Server` the need of zpages extension goes away. If we do this, then the `serviceHost` can own the entire `zpages` logic.

Reply:
> since I don't understand where these options live (do they live in service.svcSettings? are these methods part of service.Host but not part of the component.Host interface?)

Somewhere here https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/service.go#L24

Reply:
Okay, I think I get it now, and it sounds like a good idea. I created #5353 to track this. I still think this PR makes sense to merge, since it reduces the settings one needs to pass to the `service` struct , which can help us expose it.

Reply:
Based on that issue, should we actually move that into the service itself?